sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md

```diff
@@ -0,0 +1,353 @@
+                batch.tryAdd(event);
+            } catch (PayloadSizeExceededException e) {
+                System.err.println("Event is too large for batch. Exception: " + e);
+            }
+        }
+
+        return sender.send(batch);
+    });
+sendFuture.get();
+```
+
+In v5:
+```java
+List<EventData> events = Arrays.asList(EventData.create("foo".getBytes()), EventData.create("bar".getBytes()));
+
+EventHubProducerAsyncClient producer = new EventHubClientBuilder()
+    .connectionString("connection-string-for-an-event-hub")
+    .buildAsyncProducerClient();
+
+CreateBatchOptions options = new CreateBatchOptions()
+    .setPartitionId("my-partition-id");
+
+Mono<Void> sendOperation = producer.createBatch(options).flatMap(batch -> {
+    for (EventData event : data) {
+        // Assuming all events fit into a single batch. This returns false if it does not.
+        // If that is the case, we'll send the full batch then create another one to continue adding events to.
+        batch.tryAdd(event);
+    }
+    return producer.send(batch);
+});
+
+sendOperation.block();
+```
+
+### Migrating code from `PartitionReceiver` to `EventHubConsumerClient` for receiving events
+
+In V2, event handlers were passed as positional arguments to `receive`.
+
+In V5, event handlers are passed as part of a `SubscriptionEventHandlers` shaped object.
+
+For example, this code which receives from a partition in V2:
+
+```typescript
+const client = EventHubClient.createFromConnectionString(connectionString);
+const rcvHandler = client.receive(partitionId, onMessageHandler, onErrorHandler, {
+  eventPosition: EventPosition.fromStart(),
+  consumerGroup: consumerGroupName
+});
+await rcvHandler.stop();
+```
+
+Becomes this in V5:
+
+```typescript
+const eventHubConsumerClient = new EventHubConsumerClient(consumerGroupName, connectionString);
+
+const subscription = eventHubConsumerClient.subscribe(
+  partitionId, {
+ processInitialize: (initContext) => {
+ initContext.setStartingPosition(EventPosition.fromStart());
+ },
+ processEvents: onMessageHandler,
+ processError: onErrorHandler
+});
+
+await subscription.close();
+```
+
+See [`receiveEvents.ts`](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts)
+for a sample program demonstrating this.
+
+In V5, this has been consolidated into a more efficient `sendBatch` method.
+Batching merges information from multiple messages into a single send, reducing
+the amount of network communication needed vs sending messages one at a time.
+
+So in V2:
+```java
+const eventsToSend = [
+  // events go here
+];
+
+const client = EventHubClient.createFromConnectionString(connectionString);
+
+// Would fail if the total size of events exceed the max size supported by the library.
+await client.sendBatch(eventsToSend, partitionId);
+```
+
+In V5:
+```typescript
```

Comment:
nit:  I think you mean `java` here, but the snippet looks like typescript.   Is that intentional?